src/main/kotlin/com/pubnub/api/subscribe/Subscribe.kt

```diff
@@ -99,36 +104,47 @@ class Subscribe(
         withPresence: Boolean,
         withTimetoken: Long = 0L,
     ) {
+        throwExceptionIfChannelAndChannelGroupIsMissing(channels, channelGroups)
+        addChannelsToLocalStorage(channels)
+        addChannelGroupsToLocalStorage(channelGroups)
```

Comment:
This method (just like all other methods here) is not synchronized or do not guard the internal state from being modified concurrently and I would expect to see a lot of `ConcurrentModificationExceptions`. As already mentioned in other comment maybe we should think through where exactly should we handle that

Reply:
ok. I will change it.